.github/workflows/bikes.yml

```diff
@@ -13,7 +13,14 @@ jobs:
     steps:
     
     - uses: actions/checkout@master
-    
+
+    - name: Create valid child namespace name
+      id: replace
+      run: |
+        branch=$GITHUB_HEAD_REF        
+        branch=$(echo ${branch//[!0-9a-zA-Z]/-} | tr '[:upper:]' '[:lower:]' | sed '/^[-]/ s/./n/' | sed '/[-]$/ s/-$/n/')
```

Comment:
You can reduce the sed commands to a single one like `sed -e 's/^-/n/' -e 's/-$/n/'`. Also, what's the reasoning behind choosing `n` as the replacement character? It might make sense to use a lesser-used character like `z`, or even just replace the character with nothing.

Reply:
Lastly, per the Kubernetes naming rules, the value cannot be more than 63 characters in length, so make sure to trim it (easiest would be to update the first line to `branch=${GITHUB_HEAD_REF::63}`).

Reply:
For now updated to branch=${GITHUB_HEAD_REF::41} based on the current helm upgrade step. Will add a task to discuss and close on a design for this.

Reply:
Added a new step for constructing release-name from the environment variable GITHUB_HEAD_REF and honoring the helm 'release-name' spec.